src/installer/managed/Microsoft.NET.HostModel/ResourceUpdater.cs

```diff
@@ -308,6 +318,32 @@ public ResourceUpdater AddResource(byte[] data, IntPtr lpType, IntPtr lpName)
             return this;
         }
 
+        /// <summary>
+        /// Add a language-neutral integer resource from a byte[] with
+        /// a particular type and name. This will not modify the
+        /// target until Update() is called.
+        /// Throws an InvalidOperationException if Update() was already called.
+        /// </summary>
+        public ResourceUpdater AddResource(byte[] data, string lpType, IntPtr lpName)
+        {
+            if (hUpdate.IsInvalid)
+            {
+                ThrowExceptionForInvalidUpdate();
+            }
+
+            if (!IsIntResource(lpName))
+            {
+                throw new ArgumentException("AddResource can only be used with integer resource names");
```

Comment:
Not your issue, but we should reach out to someone about localizing this part of the stack. The entire SDK is localizable and that is where this is used so I would assume it falls under that charter as well.

Reply:
I think the current split is that the SDK has MSBuild tasks which consume `HostModel`. Those tasks are responsible for catching exceptions and logging them as errors with a generic localized message. And if we want more details / specialized errors, `HostModel` needs to throw different types of exception types or have different error cods (apphost does this).

Reply:
The exceptions thrown by `ResourceUpdater` are usage exceptions that indicate bugs in the code that calls it, so the SDK should never see them. In this case, I can see that this might be exposed to the SDK layer, so I'll add some new exception types that the SDK can consume when consuming the tlb work.